driver-async/src/main/com/mongodb/async/client/MongoClientImpl.java

```diff
@@ -48,7 +49,8 @@
 
     private static final RootCodecRegistry DEFAULT_CODEC_REGISTRY = new RootCodecRegistry(asList(new ValueCodecProvider(),
                                                                                                  new DocumentCodecProvider(),
-                                                                                                 new BsonValueCodecProvider()));
+                                                                                                 new BsonValueCodecProvider(),
+                                                                                                 new GridFSFileCodecProvider()));
```

Comment:
Better to figure out a way to add this provider to the registry in the MongoDatabase used by the GridFS instance.